src/main/java/io/github/syst3ms/skriptparser/lang/VariableString.java

```diff
@@ -87,6 +76,9 @@ private VariableString(Object[] data) {
             if (c == '%') {
                 if (i == charArray.length - 1) {
                     return Optional.empty();
+                } else if (charArray[i + 1] == '%') {
+                    sb.append(charArray[++i]);
```

Comment:
What about init `next` variable in top of the `for` loop and use it instead of `++i` ?

Reply:
This ultimately leads to other parts being rewritten. I will just keep it as-is currently as both options have 'up' and 'down'sides. Note that this still stays a minor inconvenience.